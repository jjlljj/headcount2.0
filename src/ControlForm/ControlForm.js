import React, { Component } from 'react';
import './ControlForm.css';

class ControlForm extends Component {
  render() {
    return (
      <div>Welcome To Headcount 2.0</div>
    );
  }
}

export default ControlForm;
