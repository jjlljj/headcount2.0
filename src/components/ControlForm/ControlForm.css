.control-form {
  align-self: center;
  margin: 10px;
}

.control-form label {
  display: none;
  margin: 10px;
}

.control-form input {
  font-size: 14px;
  margin-right: 5px;
  padding: 8px;
  width: 200px;
}