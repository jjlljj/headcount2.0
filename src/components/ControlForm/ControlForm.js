import React, { Component } from 'react';
import './ControlForm.css';

class ControlForm extends Component {
  constructor(props) {
    super(props)

    this.state = {
      location: ''
    }
  }

  handleFormInput = (e) => {
    e.preventDefault();
    const { value } = e.target
    this.setState({
      location: value
    })
    this.props.handleSearch(value)
  }

  render() {
    return (
      <form className="control-form" onSubmit={ (e)=> e.preventDefault()} >
        <label htmlFor='searchDistrict'>Find your district</label>  
        <input 
          id='searchDistrict'
          type='text'
          placeholder='Search school district'
          onChange={this.handleFormInput}
          value={this.state.location}
         />
      </form>
    );
  }
}

export default ControlForm;
