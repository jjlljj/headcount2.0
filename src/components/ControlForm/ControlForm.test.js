 /* eslint-disable */ 

import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ControlForm from './ControlForm';


describe('ControlForm', () => {

  it('blah blah blah', () => {
    
  })

});