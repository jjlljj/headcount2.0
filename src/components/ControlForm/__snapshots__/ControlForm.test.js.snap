// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ControlForm should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ControlForm />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <label
          htmlFor="searchDistrict"
>
          Find your district
</label>,
        <input
          id="searchDistrict"
          onChange={[Function]}
          placeholder="Search for a school district"
          type="text"
          value=""
/>,
      ],
      "className": "control-form",
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Find your district",
          "htmlFor": "searchDistrict",
        },
        "ref": null,
        "rendered": "Find your district",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "id": "searchDistrict",
          "onChange": [Function],
          "placeholder": "Search for a school district",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label
            htmlFor="searchDistrict"
>
            Find your district
</label>,
          <input
            id="searchDistrict"
            onChange={[Function]}
            placeholder="Search for a school district"
            type="text"
            value=""
/>,
        ],
        "className": "control-form",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Find your district",
            "htmlFor": "searchDistrict",
          },
          "ref": null,
          "rendered": "Find your district",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "id": "searchDistrict",
            "onChange": [Function],
            "placeholder": "Search for a school district",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
